spikeinterface/core/recording_tools.py

```diff
@@ -167,41 +201,66 @@ def get_chunk_with_margin(rec_segment, start_frame, end_frame,
             end_frame2 = end_frame + margin
             right_pad = 0
 
-        traces_chunk = rec_segment.get_traces(start_frame2, end_frame2, channel_indices)
-        
-        
-        if dtype is not None or window_on_margin or left_pad > 0 or right_pad > 0:
+        traces_chunk = rec_segment.get_traces(
+            start_frame2, end_frame2, channel_indices
+        )
+
+        if (
+            dtype is not None
+            or window_on_margin
+            or left_pad > 0
+            or right_pad > 0
+        ):
             need_copy = True
         else:
             need_copy = False
 
+        left_margin = margin
+        right_margin = margin
+
         if need_copy:
             if dtype is None:
                 dtype = traces_chunk.dtype
-            traces_chunk2 = np.zeros((full_size, traces_chunk.shape[1]), dtype=dtype)
-            i0 = left_pad
-            i1 = left_pad + traces_chunk.shape[0]
-            traces_chunk2[i0: i1, :] = traces_chunk
+
             left_margin = margin
             if end_frame < (length + margin):
                 right_margin = margin
             else:
                 right_margin = end_frame + margin - length
-            if window_on_margin:
-                # apply inplace taper  on border
-                taper = (1 - np.cos(np.arange(margin) / margin * np.pi)) / 2
-                taper = taper[:, np.newaxis]
-                traces_chunk2[:margin] *= taper
-                traces_chunk2[-margin:] *= taper[::-1]
-            traces_chunk = traces_chunk2
-        else:
-            left_margin = margin
-            right_margin = margin
+
+            if add_zeros:
+                traces_chunk2 = np.zeros(
+                    (full_size, traces_chunk.shape[1]), dtype=dtype
+                )
+                i0 = left_pad
+                i1 = left_pad + traces_chunk.shape[0]
+                traces_chunk2[i0:i1, :] = traces_chunk
+                if window_on_margin:
+                    # apply inplace taper on border
+                    taper = (
+                        1 - np.cos(np.arange(margin) / margin * np.pi)
+                    ) / 2
+                    taper = taper[:, np.newaxis]
+                    traces_chunk2[:margin] *= taper
+                    traces_chunk2[-margin:] *= taper[::-1]
+                traces_chunk = traces_chunk2
+            elif add_reflect_padding:
+                # in this case, we don't want to taper
```

Comment:
Maybe add a waord in the doc about this, no ?

Reply:
Added something, thanks!